experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -736,35 +742,32 @@ function compareCellPositions(
 	baseIntention: RevisionTag | undefined,
 	baseMark: EmptyInputCellMark<unknown>,
 	newMark: EmptyInputCellMark<unknown>,
-	gapOffsetInBase: number,
 ): number {
 	const baseId = getCellId(baseMark, baseIntention);
+	const baseLength = getMarkLength(baseMark);
 	assert(baseId !== undefined, 0x6a0 /* baseMark should have cell ID */);
+	const baseIdLast = (baseId.id as number) + baseLength - 1;
 	const newId = getCellId(newMark, undefined);
+	const newLength = getMarkLength(newMark);
 	if (baseId.revision === newId?.revision) {
-		return baseId.index - newId.index;
+		const newIdLast = (newId.id as number) + newLength - 1;
+		if (baseId.id >= newId.id && baseId.id <= newIdLast) {
+			return baseId.id - newId.id;
+		} else if (newId.id >= baseId.id && newId.id <= baseIdLast) {
+			return baseId.id - newId.id;
+		}
 	}
```

Comment:
Nit: I find this easier to read:
```suggestion
		if (newId.id <= baseId.id && baseId.id <= newIdLast) {
			return baseId.id - newId.id;
		} else if (baseId.id <= newId.id && newId.id <= baseIdLast) {
			return baseId.id - newId.id;
		}
```
You could also unify the return statements:
```suggestion
		if ((newId.id <= baseId.id && baseId.id <= newIdLast) || (baseId.id <= newId.id && newId.id <= baseIdLast)) {
			return baseId.id - newId.id;
		}
```

Reply:
I just noticed there is a `areOverlappingIdRanges` function the `utils.ts` file that you could use here.